api/envoy/config/tap/v3/common.proto

```diff
@@ -101,6 +107,30 @@ message HttpHeadersMatch {
   repeated route.v3.HeaderMatcher headers = 1;
 }
 
+// HTTP generic body match configuration.
+// List of text strings and hex strings to be located in HTTP body.
+// All specified strings must be found in the HTTP body for positive match.
+// The search may be limited to specified number of bytes from the body start.
+message HttpGenericBodyMatch {
+  message GenericTextMatch {
+    oneof rule {
+      option (validate.required) = true;
+
+      // Text string to be located in HTTP body.
+      string contains_text = 1;
+
+      // Text with hex digits to be located in HTTP body.
+      string contains_hex = 2;
```

Comment:
Should this be proto3 `bytes`? The JSON/YAML representations allow for base64.

Reply:
Changed names to `string_match` and `binary_match`. Investigating using `bytes` as type for `binary_match`.

Reply:
@htuch I need your approval on API changes, please.